### PR TITLE
Add new recipe for ingrid package

### PR DESCRIPTION
**Pull request by smithsp**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

I had to patch the setup.py to avoid the use of `setuptools-git-version`.

Mention @xinganthony @kalling @orso82 who may want to be maintainers as well

### CODE REVIEW

**Comment by smithsp** [comment]
Linter and checks passed here @conda-forge/help-python

**Comment by xinganthony** [comment]
Thanks for your work @smithsp !

Pinging: @bryan-garcia and @umansky: We are making a conda package of INGRID to ease its installation and integration in OMFIT. Please let @smithsp know if you would like to be maintainers.

File: recipes/ingrid/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  patches:
+    - setup.py.patch
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
```
Comment (marcelotrevisani) [inline recipes/ingrid/meta.yaml:25]:
```suggestion
  host:
```

File: recipes/ingrid/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 14 is convenient
+# when copying and pasting from another recipe, but not really needed.
```
Comment (marcelotrevisani) [inline recipes/ingrid/meta.yaml:5]:
```suggestion
```

File: recipes/ingrid/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  build:
+    - python >=3.6,<4
+    - pip
+    - setuptools
+    - sympy
+    - m2r2
+    - scipy >=1.3.1
+    - numpy
+    - matplotlib
+    - pyyaml
```
Comment (marcelotrevisani) [inline recipes/ingrid/meta.yaml:34]:
```suggestion
    - pip
    - setuptools
```

File: recipes/ingrid/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  run:
+    - python >=3.6,<4
+    - numpy
+    - sympy
+    - m2r2
+    - scipy >=1.3.1
+    - matplotlib
+    - pyyaml
```
Comment (marcelotrevisani) [inline recipes/ingrid/meta.yaml:42]:
```suggestion
    - python >=3.6,<4
    - numpy
    - sympy
    - m2r2
    - scipy >=1.3.1
    - matplotlib-base
    - pyyaml
```

File: recipes/ingrid/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+test:
+  # Some packages might need a `test/commands` key to check CLI.
+  # List all the packages/modules that `run_test.py` imports.
```
Comment (marcelotrevisani) [inline recipes/ingrid/meta.yaml:46]:
```suggestion
```

File: recipes/ingrid/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  imports:
+    - INGRID
+#    - simplejson.tests
+  # For python packages, it is useful to run pip check. However, sometimes the
+  # metadata used by pip is out of date. Thus this section is optional if it is
+  # failing.
```
Comment (marcelotrevisani) [inline recipes/ingrid/meta.yaml:52]:
```suggestion
```

File: recipes/ingrid/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  requires:
+    - matplotlib
+    - numpy
+#  commands:
+#    - pip check
```
Comment (marcelotrevisani) [inline recipes/ingrid/meta.yaml:57]:
```suggestion
    - pip
  commands:
    - pip check
```

File: recipes/ingrid/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+about:
+  home: https://github.com/LLNL/INGRID
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: 'Tokamak edge plasma grid generator for magnetic-topologies with up to two x-points anywhere in the domain.'
+  # The remaining entries in this section are optional, but recommended.
```
Comment (marcelotrevisani) [inline recipes/ingrid/meta.yaml:65]:
```suggestion
```

File: recipes/ingrid/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    - pip
+    - setuptools
+  run:
+    - python >=3.6,<4
+    - numpy
+    - sympy
+    - m2r2
+    - scipy >=1.3.1
+    - matplotlib-base
+    - pyyaml
```
Comment (marcelotrevisani) [inline recipes/ingrid/meta.yaml:31]:
```suggestion
    - pyyaml
    - tk
```

File: recipes/ingrid/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 1d92d669aabb72f2a83763c28a8ad682cc530ae47f11e7131afeda025efd286c
+  patches:
+    - setup.py.patch
+
+build:
+  noarch: python
```
Comment (smithsp) [inline recipes/ingrid/meta.yaml:15]:
```suggestion
```

File: recipes/ingrid/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 1d92d669aabb72f2a83763c28a8ad682cc530ae47f11e7131afeda025efd286c
+  patches:
+    - setup.py.patch
+
+build:
+  number: 0
```
Comment (smithsp) [inline recipes/ingrid/meta.yaml:15]: Add back noarch
```suggestion
build:
  number: 0
  noarch: python
```

File: recipes/ingrid/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    - scipy >=1.3.1
+    - matplotlib-base
+    - pyyaml
+    - tk
+
+test:
+  imports:
+    - INGRID
+  requires:
+    - pip
```
Comment (smithsp) [inline recipes/ingrid/meta.yaml:37]:
```suggestion
    - pip
    - tk
    - matplotlib-base
    - scipy
    - numpy
    - sympy
```

**Review by smithsp — COMMENTED** [review]
Get testing requirements straight

File: recipes/ingrid/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    - tk
+    - matplotlib-base
+    - scipy
+    - numpy
+    - sympy
+  commands:
+    - pip check
```
Comment (smithsp) [inline recipes/ingrid/meta.yaml:45]:
```suggestion
```

File: recipes/ingrid/setup.py.patch [hunk 14]
```diff
@@ -0,0 +1,11 @@
+--- setup.py.orig	2021-06-04 13:34:15.796271806 -0700
++++ setup.py	2021-06-04 13:34:23.408238740 -0700
+@@ -20,8 +20,6 @@
+ 
+ setup(
+     name='INGRID',
+-    version_format='{tag}',
+-    setup_requires=['setuptools-git-version'],
```
Comment (chrisburr) [inline recipes/ingrid/setup.py.patch:8]: The version isn't being set. This is probably the easiest workaround though ideally the upstream package should use setuptools_scm instead:
```suggestion
-    setup_requires=['setuptools-git-version'],
+    version=os.environ["PKG_VERSION"],
```

File: recipes/ingrid/setup.py.patch [hunk 15]
```diff
@@ -0,0 +1,11 @@
+--- setup.py.orig	2021-06-04 13:34:15.796271806 -0700
++++ setup.py	2021-06-04 13:34:23.408238740 -0700
+@@ -20,8 +20,6 @@
+ 
+ setup(
```
Comment (chrisburr) [inline recipes/ingrid/setup.py.patch:5]:
```suggestion
+ import os
  setup(
```

File: recipes/ingrid/meta.yaml [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+    - sympy
+    - m2r2
+    - scipy >=1.3.1
+    - matplotlib-base
+    - pyyaml
+    - tk
+
+test:
+  imports:
+    - INGRID
```
Comment (chrisburr) [inline recipes/ingrid/meta.yaml:36]:
```suggestion
    - INGRID
  commands:
    - pip check
  requires:
    - pip
```

Comment (smithsp) [reply recipes/ingrid/meta.yaml:36]: @chrisburr Thanks for the review.  Despite listing `tk` as a conda dependency, `pip check` uses a different method of checking for the `tk` package, and indeed `pip install tk` does install something.  So I disabled the `pip check`.  Should I put a patch on the `setup.py` file instead to remove the `tk` from the dependencies?

File: recipes/ingrid/setup.py.patch (hunk 14 above)
Comment (smithsp) [reply recipes/ingrid/setup.py.patch:8]: I will put this in the patch once a decision is made about how to handle the `tk` pip requirement.

File: recipes/ingrid/meta.yaml (hunk 16 above)
Comment (chrisburr) [reply recipes/ingrid/meta.yaml:36]: I think `tk` in the `setup.py` dependencies isn't installing what you think it is:
* https://github.com/LLNL/INGRID/blob/78e0f278effd37b6860032e830344041ed5702e1/setup.py#L55
* https://pypi.org/project/tk/

I guess you want https://docs.python.org/3/library/tkinter.html which is part of the standard library.

**Pull request closed without merging**